napari/components/_tests/test_add_layers.py

```diff
@@ -35,6 +35,21 @@ def test_add_layers_with_plugins(layer_datum):
         assert all(lay.source == expected_source for lay in v.layers)
 
 
+def test_add_layers_with_plugins_full_layers(layer_data_and_types):
+    """Test that add_layers_with_plugins works for full Layer objects."""
+    layers, _, _, layer_fnames = layer_data_and_types
+    for layer, layer_fname in zip(layers, layer_fnames):
+        with patch(
+            'napari.plugins.io.read_data_with_plugins',
+            MagicMock(return_value=([layer], _testimpl)),
+        ):
+            v = ViewerModel()
+            v._add_layers_with_plugins([layer_fname], stack=False)
+            assert len(v.layers) == 1
+            assert v.layers[0].source.path == layer_fname
+            assert v.layers[0].source.reader_plugin == 'testimpl'
```

Comment:
I do not understand for loop and asserting on `v.layers[0]`

Reply:
The for loop was iterating through the different layers returned by the fixture. I realized though that this fixture only had a couple of `Image` and `Points` layers so I've swapped it to use the `layer` fixture now which covers all layer types. 

We still assert `viewer.layers[0]` to make sure the layer added to the viewer has the correct source set.